aws/resource_aws_codebuild_webhook.go

```diff
@@ -0,0 +1,112 @@
+package aws
+
+import (
+	"github.com/aws/aws-sdk-go/aws"
+	"github.com/aws/aws-sdk-go/service/codebuild"
+	"github.com/hashicorp/terraform/helper/schema"
+)
+
+func resourceAwsCodeBuildWebhook() *schema.Resource {
+	return &schema.Resource{
+		Create: resourceAwsCodeBuildWebhookCreate,
+		Read:   resourceAwsCodeBuildWebhookRead,
+		Delete: resourceAwsCodeBuildWebhookDelete,
+		Update: resourceAwsCodeBuildWebhookUpdate,
+
+		Schema: map[string]*schema.Schema{
+			"name": {
+				Type:     schema.TypeString,
+				Required: true,
+				ForceNew: true,
+			},
+			"branch_filter": {
+				Type:     schema.TypeString,
+				Optional: true,
+			},
+			"url": {
+				Type:     schema.TypeString,
+				Computed: true,
+			},
+		},
+	}
+}
+
+func resourceAwsCodeBuildWebhookCreate(d *schema.ResourceData, meta interface{}) error {
+	conn := meta.(*AWSClient).codebuildconn
+
+	resp, err := conn.CreateWebhook(&codebuild.CreateWebhookInput{
+		ProjectName:  aws.String(d.Get("name").(string)),
+		BranchFilter: aws.String(d.Get("branch_filter").(string)),
+	})
+	if err != nil {
+		return err
+	}
+
+	d.SetId(d.Get("name").(string))
+	d.Set("branch_filter", d.Get("branch_filter").(string))
+	d.Set("url", resp.Webhook.Url)
+	return nil
+}
+
+func resourceAwsCodeBuildWebhookRead(d *schema.ResourceData, meta interface{}) error {
+	conn := meta.(*AWSClient).codebuildconn
+
+	resp, err := conn.BatchGetProjects(&codebuild.BatchGetProjectsInput{
+		Names: []*string{
+			aws.String(d.Id()),
+		},
+	})
+
+	if err != nil {
+		return err
+	}
+
+	if len(resp.Projects) == 0 {
+		d.SetId("")
```

Comment:
When we're removing a resource from the state, we should log an appropriate warning message in the logs, e.g.

```go
log.Printf("[WARN] CodeBuild Project %q not found, removing from state", d.Id())
```